### PR TITLE
Bump XGBoost to 1.2 and Sagemaker SDK to v2 in XGBoost abalone notebooks

### PR DESCRIPTION
**Pull request by edwardjkim**
*Description of changes:*

Revision 2 (12/7/2020)

Following https://github.com/aws/amazon-sagemaker-examples/pull/1713, `pip install` and `pip uninstall` have been removed. The notebooks have been tested on Studio. I confirmed that they work without `pip install` and added text to the notebook along the line of "This notebook was tested in Amazon SageMaker Studio on a ml.t3.medium instance with Python 3 (Data Science) kernel."

Revision 1

Bumping XGBoost version to 1.2-1. Also using SageMaker Python SDK v2.x since XGBoost 1.2-1 is not supported in v1. Corresponding PR for making XGBoost 1.2-1 available in SDK v2 (https://github.com/aws/sagemaker-python-sdk/pull/1899) has been merged and released to PyPI, and this PR is ready to be merged.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by edwardjkim** [comment]
@iyerr3 Python SDK v2 now has image URIs for 1.2-1 and the notebooks are ready. Please review and merge if they look okay (I don't have merge permissions).

File: introduction_to_amazon_algorithms/xgboost_abalone/xgboost_abalone_dist_script_mode.ipynb [hunk 1]
```diff
@@ -48,7 +48,7 @@
    "outputs": [],
    "source": [
     "import sys\n",
-    "!{sys.executable} -m pip install -qU awscli boto3 \"sagemaker>=1.71.0,<2.0.0\""
+    "!{sys.executable} -m pip install -qU awscli boto3 \"sagemaker==2.*\""
```
Comment (aaronmarkham) [inline introduction_to_amazon_algorithms/xgboost_abalone/xgboost_abalone_dist_script_mode.ipynb:51]: Please refer the [template for upgrade/downgrade approach](https://github.com/aws/amazon-sagemaker-examples/blob/master/template.ipynb) that won't mess up the kernel.

**Review by aaronmarkham — CHANGES_REQUESTED** [review]
Please refer the template for upgrade/downgrade approach that won't mess up the kernel.
Also, it would help to make the required version of SDK very clear in the intro/prerequisites section of the notebooks.

File: introduction_to_amazon_algorithms/xgboost_abalone/inference.py [hunk 2]
```diff
@@ -51,16 +51,16 @@ def predict_fn(input_data, model):
...
 
-def output_fn(prediction, content_type):
+def output_fn(predictions, content_type):
     """
     After invoking predict_fn, the model server invokes `output_fn`.
     """
-    if content_type == "application/json":
-        return json.dumps(prediction.tolist())
+    if content_type == "text/csv":
+        return ','.join(str(x) for x in predictions[0])
```
[the remark this answers is not in the document]
Comment (edwardjkim) [reply introduction_to_amazon_algorithms/xgboost_abalone/inference.py:64]: The default content type for the XGBoost estimator was changed to `text/csv` in SDK v2. JSON is still supported but it requires overriding the content type by passing an additional argument to the estimator, and I think changing the content type in the script is simpler.

File: introduction_to_amazon_algorithms/xgboost_abalone/xgboost_inferenece_script_mode.ipynb [hunk 3]
```diff
@@ -47,7 +47,32 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "!{sys.executable} -m pip install -qU awscli boto3 \"sagemaker==2.*\""
```
[the remark this answers is not in the document]
Comment (edwardjkim) [reply introduction_to_amazon_algorithms/xgboost_abalone/xgboost_inferenece_script_mode.ipynb]: Thanks for catching this. Removed.

File: introduction_to_amazon_algorithms/xgboost_abalone/xgboost_parquet_input_training.ipynb [hunk 4]
```diff
@@ -47,7 +63,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "!python -m pip install pyarrow==0.15"
+    "!python -m pip install pyarrow==0.16  # upgrade to 0.16 for xgboost 1.2-1"
```
[the remark this answers is not in the document]
Comment (edwardjkim) [reply introduction_to_amazon_algorithms/xgboost_abalone/xgboost_parquet_input_training.ipynb]: Changed to `%pip`.

**Comment by edwardjkim** [comment]
@aaronmarkham Sorry about the delayed follow up. I was waiting for SDK v2 to become available on Studio and/or Mead, and now that it's available, I've tested the the notebooks again on Studio and made some changes accordingly. Could you take another look when you get a chance? Thanks.

**Review by aaronmarkham — APPROVED** [review]

**Pull request merged**